python/cugraph/structure/shuffle.py

```diff
@@ -0,0 +1,46 @@
+import math
+from dask.dataframe.shuffle import rearrange_by_column
+
+
+def get_n_workers():
+    from dask.distributed import default_client
+    client = default_client()
+    return len(client.scheduler_info()['workers'])
+
+
+def get_2D_div(ngpus):
+    pcols = int(math.sqrt(ngpus))
+    while ngpus % pcols != 0:
```

Comment:
Too lazy today to sit down and do math, but we should be able to directly compute this without a while loop.